### DISCUSSION TITLE
What are the alternatives to setNativeProps in fabric?

### DISCUSSION THREAD

**Discussion by rushatgabhane** [reactwg/react-native-new-architecture#77] [Q&A] [answered]
The [docs](https://reactnative.dev/docs/new-architecture-library-intro#migrating-off-setnativeprops) say that we should migrate all values of `setNativeProps` to state. This is painful because it has a performance impact due to re-rendering of components.

Is there any other alternative?

**Comment by sammy-SC** [comment, marked as answer]
Hello @rushatgabhane,

what's your use case? Where are you running into performance issues?

We were able to migrate all uses of `setNativeProps` to use state. `setNativeProps` breaks model for the new architecture in a fundamental way and there wasn't really a way to support it moving forward. `setNativeProps` achieves its performance benefits by skipping all steps. This way, we get the performance benefit, but representation of UI gets out of sync between React, React Native and what's actually shown to the user. It limits our ability to add new features like synchronous rendering or unified animations.
